File: akka-serialization-jackson/src/test/scala/doc/akka/serialization/jackson/v2a/Customer.scala
/*
 * Copyright (C) 2019-2025 Lightbend Inc. <https://www.lightbend.com>
 */

package doc.akka.serialization.jackson.v2a

import akka.serialization.jackson.JsonSerializable

// #structural
case class Customer(name: String, shippingAddress: Address, billingAddress: Option[Address]) extends JsonSerializable
// #structural

File: akka-serialization-jackson/src/test/java/jdoc/akka/serialization/jackson/v2a/Customer.java
/*
 * Copyright (C) 2019-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package jdoc.akka.serialization.jackson.v2a;

import akka.serialization.jackson.JsonSerializable;
import java.util.Optional;

// #structural
public class Customer implements JsonSerializable {
  public final String name;
  public final Address shippingAddress;
  public final Optional<Address> billingAddress;

  public Customer(String name, Address shippingAddress, Optional<Address> billingAddress) {
    this.name = name;
    this.shippingAddress = shippingAddress;
    this.billingAddress = billingAddress;
  }
}
// #structural
